src/python/pants/engine/legacy/graph.py

```diff
@@ -734,7 +735,9 @@ def _eager_fileset_with_spec(
     Get[Owners](OwnersRequest(sources=snapshot.files)) for snapshot in snapshot_per_include
   )
   result: List[AddressWithOrigin] = []
-  for spec, owners in zip(filesystem_specs.includes, owners_per_include):
+  for spec, snapshot, owners in zip(
+    filesystem_specs.includes, snapshot_per_include, owners_per_include
+  ):
```

Comment:
This is the heart of the change. We were originally throwing away the information gained from `snapshot`, i.e. which files were actually resolved. Now, we preserve `snapshot.files`.